### ISSUE TITLE
HF Down, Axo Dead (SOLVED: TRANSFORMERS_OFFLINE=1 env)

### ISSUE THREAD

**Issue by fblgit** [axolotl-ai-cloud/axolotl#1285]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

Axolotl works with local model and local dataset without HF hub or HF at all.

### Current behaviour

There is a `whoami` and who knows what else that calls HF. When HF was down, train couldnt start.

### Steps to reproduce

/etc/hosts entry for huggingface.co hostnames
try to train anything

### Possible solution

capture exceptions for HF calls, None those objects and cascade it accordingly.

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10

### axolotl branch-commit

main

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by winglian** [comment]
You can set TRANSFORMERS_OFFLINE=1 in your env vars

**Comment by fblgit** [comment]
wah.. didnt knew that one :D i saw anyways in the code the `whoami`. I'll give it a shot with this env_var.
thank you!

**Comment by winglian** [comment]
You probably also need HF_DATASETS_OFFLINE=1.

**Comment by ron0studios** [comment]
Hi, apologies for reopening this. I have followed the steps above and am still facing the same issue. 
Here is my yml file right now (based on the examples/Llama-3/lora-1b.yml)
```yml
base_model: ./llama_model
# NousResearch/Llama-3.2-1B
# Automatically upload checkpoint and final model to HF
# hub_model_id: username/custom_model_name

load_in_8bit: false
load_in_4bit: false
strict: false

datasets:
  - path: ./mental_health_counseling_conversations/combined_dataset.json
    ds_type: json
    type:
      system_prompt: ""
      field_system: system
      field_instruction: Context
      field_output: Response
      format: "[INST] {instruction} [/INST]"
      no_input_format: "[INST] {instruction} [/INST]"


dataset_prepared_path: last_run_prepared
val_set_size: 0.1
output_dir: ./outputs/lora-out

adapter: lora
lora_model_dir:

sequence_len: 2048
sample_packing: true
eval_sample_packing: true
pad_to_sequence_len: true

lora_r: 16
lora_alpha: 32
lora_dropout: 0.05
lora_fan_in_fan_out:
lora_target_modules:
  - gate_proj
  - down_proj
  - up_proj
  - q_proj
  - v_proj
  - k_proj
  - o_proj

wandb_project:
wandb_entity:
wandb_watch:
wandb_name:
wandb_log_model:

gradient_accumulation_steps: 2
micro_batch_size: 2
num_epochs: 1
optimizer: adamw_8bit
lr_scheduler: cosine
learning_rate: 0.0002

train_on_inputs: false
group_by_length: false
bf16: auto
fp16:
tf32: false

gradient_checkpointing: true
early_stopping_patience:
resume_from_checkpoint:
local_rank:
logging_steps: 1
xformers_attention:
flash_attention: true

loss_watchdog_threshold: 5.0
loss_watchdog_patience: 3

warmup_steps: 10
evals_per_epoch: 4
saves_per_epoch: 1
debug:
deepspeed:
weight_decay: 0.0
fsdp:
fsdp_config:
special_tokens:
  pad_token: "<|end_of_text|>"
```
